src/Generator/FreeFormSchemaGenerator.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+declare(strict_types=1);
+
+namespace DoclerLabs\ApiClientGenerator\Generator;
+
+use DoclerLabs\ApiClientGenerator\Entity\Constraint\ConstraintCollection;
+use DoclerLabs\ApiClientGenerator\Entity\Field;
+use DoclerLabs\ApiClientGenerator\Entity\FieldType;
+use DoclerLabs\ApiClientGenerator\Input\Specification;
+use DoclerLabs\ApiClientGenerator\Output\Php\PhpFileCollection;
+use JsonSerializable;
+use PhpParser\Node\Stmt\ClassMethod;
+use stdClass;
+
+class FreeFormSchemaGenerator extends MutatorAccessorClassGeneratorAbstract
+{
+    private const FREE_FORM_SCHEMA_VARIABLE = 'data';
+
+    public function generate(Specification $specification, PhpFileCollection $fileRegistry): void
+    {
+        $compositeFields = $specification->getCompositeFields()->getUniqueByPhpClassName();
+        foreach ($compositeFields as $field) {
+            if ($field->isFreeFormObject()) {
+                $this->generateFreeFormSchema($field, $fileRegistry);
+            }
+        }
+    }
+
+    private function generateFreeFormSchema(Field $field, PhpFileCollection $fileRegistry): void
+    {
+        $this->addImport(JsonSerializable::class);
+        $this->addImport(stdClass::class);
+
+        $className     = $field->getPhpClassName();
+        $freeFormField = $this->generateFreeFormField();
+
+        $classBuilder = $this->builder
+            ->class($className)
+            ->implement('SerializableInterface', 'JsonSerializable')
+            ->addStmt($this->generateProperty($freeFormField))
+            ->addStmt($this->generateConstructor())
+            ->addStmt($this->generateGet($freeFormField))
+            ->addStmt($this->generateToArray())
+            ->addStmt($this->generateJsonSerialize());
+
+        $this->registerFile($fileRegistry, $classBuilder, SchemaGenerator::SUBDIRECTORY, SchemaGenerator::NAMESPACE_SUBPATH);
+    }
+
+    private function generateConstructor(): ClassMethod
+    {
+        $param = $this->builder
+            ->param(self::FREE_FORM_SCHEMA_VARIABLE)
+            ->setType(FieldType::PHP_TYPE_ARRAY)
+            ->getNode();
+
+        $paramInit = $this->builder->assign(
+            $this->builder->localPropertyFetch(self::FREE_FORM_SCHEMA_VARIABLE),
+            $this->builder->castToObject(
```

Comment:
Why do we have this casting here? Why not only in the getter method? 
basically there is an array, then object, then array again for `toArray`

Reply:
Because it is an object - so I thought it is good to keep it as the object in the basic state. So I made two castings. And you are right that it can be only one casting in the get() function.

Reply:
I tried to apply your proposal and now I would skip this because of the typehint complications - I have to create fake field with changed type and probably it won't add cleanliness and readability to the code